### PR TITLE
Add support for MLS External Add Proposals

### PR DESCRIPTION
**Pull request by smatting**
https://wearezeta.atlassian.net/browse/FS-435

## Checklist

 - [x] The **PR Title** explains the impact of the change.
 - [x] The **PR description** provides context as to why the change should occur and what the code contributes to that effect. This could also be a link to a JIRA ticket or a Github issue, if there is one.
 - [x] **changelog.d** contains the following bits of information ([details](https://github.com/wireapp/wire-server/blob/develop/docs/developer/changelog.md)):
   - [x] A file with the changelog entry in one or more suitable sub-sections. The sub-sections are marked by directories inside `changelog.d`.

### CODE REVIEW

**Comment by stefanwire** [comment]
Tests fail due to pending https://github.com/wireapp/mls-test-cli/pull/3.

File: services/galley/test/integration/API/MLS.hs [hunk 1]
```diff
@@ -1384,3 +1390,107 @@ propInvalidEpoch = withSystemTempDirectory "mls" $ \tmp -> do
...
+            tmp </> "group",
+            tmp </> "welcome"
+          ]
+      )
+      Nothing
+
+  bobClient2Qid <-
+    userClientQid (pUserId bob)
+      <$> withLastPrekeys (setupUserClient tmp CreateWithKey True (pUserId bob))
+  externalProposal <- liftIO $ createExternalProposal tmp bobClient2Qid "group" "group"
```
Comment (stephen-smith) [inline services/galley/test/integration/API/MLS.hs:1423]: Feels like this, at least, is duplicated 2+ times.  You sure it's not worth a utility function?

**Review by stephen-smith — APPROVED** [review]

File: services/galley/test/integration/API/MLS.hs (hunk 1 above)
Comment (stefanwire) [reply services/galley/test/integration/API/MLS.hs:1423]: Valid point! 🤔

**Comment by smatting** [comment]
Looks good to me!

**Review by stephen-smith — APPROVED** [review]

**Pull request merged**